docs/DeveloperGuide.md

```diff
@@ -5,9 +5,238 @@
+
+![CaloriesAddEntrySeqDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/RexDG/docs/CaloriesAddEntrySeqDiagram.puml)
+
+### Calculating calorie requirements based on a user`s goals
+
+#### Implementation
+
+This functionality is facilitated by `UserGoals`. It implements one operation, namely:
+- `UserGoals#getHealthInfo(User)`
+
+This operation is exposed in the `User` class as `User#getHealhInfo()`.
+
+Given below is an example usage scenario and how this mechanism behaves at every step:
+- Step 1: When the user inputs the command `user setup` in the terminal, 
+  the input string is sent to `Ui#handleUserCommands(User,String)`, which calls `User#setup(String)`.
+
+- Step 2: `User#setup(String)` calls `ParserUser#parseSetUp(String, User)` which parses the user's information such as 
+  his height, weight, age, gender, exercise levels and intended goal and sets these information to the `User`.
+
+- Step 3: `User#setup(String)` then calls `User#getHealthInfo()` which calls `UserGoals#getHealthInfo(User)`.
+
+- Step 4: This method calculates the basal metabolism rate of the `user` using the user's height, weight, age and gender.
+
+- Step 5: Using the `user`'s exercise levels, the method calculates his active metabolism rate, and finally calculates the calories required by the user to achieve their goals based on their input.
+
+- Step 6: This value is set to `User.caloriesRequired` by `User#setCaloriesRequired(int)`.
+
+The Sequence Diagram for the above-mentioned process is as follows:
+![Sequence Diagram](UserCalculateCaloriesSeqDiagram.png)
+
+
+#### Design considerations
+
+- **Alternative 1 (current choice):** Uses an algorithm to find the number of calories needed
+  - Pros: Not dependent on external APIs
+  - Cons: Need to come up with an algorithm to use
+
+- **Alternative 2 :** Uses an API to get the calories needed
+    - Pros: No need to figure out the optimal algorithm
+    - Cons: Need to parse response to sieve out necessary information
+
+### Calories list feature
+
+The `calories list` feature lists out the record of all the Calories data that the user has keyed in. The Calories data are all stored into a `ArrayList<Entry> calorieArrayList` attribute of the `CalorieList` Class. Calories data are printed when the `printCalorieList()` function is called. 
+
+The `printCalorieList()` function iterates through the `calorieArrayList` and prints out the Entries according to its order in the Array List.
+
+The Class diagram and sequence diagram for Calories list feature is shown below. Unrelated attributes and Classes were excluded.
+
+![CaloriesListClassDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/DevGuideRex/docs/CaloriesListClassDiagram.puml)
+![CaloriesListSequenceDiagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/rexyyong/tp/RexDG/docs/CaloriesListSequenceDiagram.puml)
```

Comment:
A few points to note:

1. Im not sure about what you are trying to convey in this diagram. The `get` activation bar seems to extend beyond the `loop` frame, which implies that the act of `get`ting breaks out of the loop?

![image](https://github.com/nus-cs2113-AY2324S2/tp/assets/69704055/ee4bbda5-e294-47f0-b687-7c02882ceaa8)

2. Missing colon for sequence diagram